knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -26,13 +27,20 @@ var (
 	log               = common.Log.WithName("consoleclidownload")
 )
 
-// Apply installs kn ConsoleCLIDownload and its required resources
-func Apply(instance *operatorv1beta1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
+// Apply installs kn ConsoleCLIDownload and its required resources when applicable
+func Apply(instance *operatorv1beta1.KnativeServing, apiclient client.Client, apiExtensionV1Client apiextensionv1.ApiextensionsV1Interface) error {
+	// Install the cli download route even if there is no console available to allow fetching the related binaries
 	route, err := reconcileKnConsoleCLIDownloadRoute(apiclient, instance)
 	if err != nil {
 		return err
 	}
-
+	// Skip installing console cli download resources if there are no related CRDs available eg. cluster is installed without console
+	if _, err = apiExtensionV1Client.CustomResourceDefinitions().Get(context.Background(), "consoleclidownloads.console.openshift.io", metav1.GetOptions{}); err != nil {
```

Comment:
Nit: Maybe a variable for `"consoleclidownloads.console.openshift.io"` to use here and in tests?